pydoctor/epydoc2stan.py

```diff
@@ -82,6 +82,15 @@ def look_for_intersphinx(self, name: str) -> Optional[str]:
         """
         return self.obj.system.intersphinx.getLink(name)
 
+    def resolve_identifier(self, identifier: str) -> Optional[str]:
+        fullID = self.obj.expandName(identifier)
+
+        target = self.obj.system.objForFullName(fullID)
+        if target is not None:
+            return target.url
```

Comment:
it looks like there is not test for this line.

Reply:
Correct: the `resolve_identifier()` method is implemented by the test helper class `RecordingAnnotationLinker` in the new tests, instead of the implementation in `_EpydocLinker`. And the old tests don't happen to trigger this line. But the old tests don't validate any of the method, so the other lines aren't really tested either, even though they execute.

Reply:
Maybe some of the old tests that call `resolve_identifier_xref()` could be adapted to call and verify this method as well without too many changes; I'll have a look.

Reply:
Partial success: it was fairly easy to adapt the old tests to also call the new method, but the old tests didn't cover the case where an identifier is found within the code being documented.

Reply:
I've written a new test case to test a successful internal match by full name.